scripts/ccpp_track_variables.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python
+
+# Standard modules
+import argparse
+import logging
+import collections
+import glob
+
+# CCPP framework imports
+from metadata_table import find_scheme_names, parse_metadata_file
+from ccpp_prebuild import import_config, gather_variable_definitions
+from mkstatic import Suite
+from parse_checkers import registered_fortran_ddt_names
+
+###############################################################################
+# Set up the command line argument parser and other global variables          #
+###############################################################################
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-s', '--sdf',           action='store', \
+                    help='suite definition file to parse', required=True)
+parser.add_argument('-m', '--metadata_path', action='store', \
+                    help='path to CCPP scheme metadata files', required=True)
+parser.add_argument('-c', '--config',        action='store', \
+                    help='path to CCPP prebuild configuration file', required=True)
+parser.add_argument('-v', '--variable',      action='store', \
+                    help='variable to track through CCPP suite', required=True)
+#parser.add_argument('--draw',  action='store_true', \
+#                    help='draw graph of calling tree for given variable', default=False)
+parser.add_argument('--debug', action='store_true', help='enable debugging output', default=False)
+args = parser.parse_args()
+
+###############################################################################
+# Functions and subroutines                                                   #
+###############################################################################
+
+def parse_arguments(args):
+    """Parse command line arguments."""
+    success = True
+    sdf = args.sdf
+    var = args.variable
+    configfile = args.config
+    metapath = args.metadata_path
+    debug = args.debug
+    return(success,sdf,var,configfile,metapath,debug)
+
+def setup_logging(debug):
+    """Sets up the logging module and logging level."""
+    success = True
+    if debug:
+        level = logging.DEBUG
+    else:
+        level = logging.WARNING
+    logging.basicConfig(format='%(levelname)s: %(message)s', level=level)
+    if debug:
+        logging.info('Logging level set to DEBUG')
+    return success
+
+def parse_suite(sdf):
+    """Reads the provided sdf, parses into a Suite data structure, including the "call tree": 
+       the ordered list of schemes for the suite specified by the provided sdf"""
+    logging.info(f'Reading sdf {sdf} and populating Suite object')
+    suite = Suite(sdf_name=sdf)
+    success = suite.parse()
+    if not success:
+        logging.error(f'Parsing suite definition file {sdf} failed.')
+        success = False
+        return (success, suite)
```

Comment:
This is a limitation of how errors are handled in the existing object definitions; since they handle errors through a "success" flag I don't want to try to mess with that, especially since this code will be superseded in the (hopefully) near-future and need updating regardless.